### PR TITLE
implement EnqueueExtensions interface in serviceaffinity

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind feature
/sig scheduling

#### What this PR does / why we need it:

Implement EnqueueExtensions interface in serviceaffinity. See the inline comments for more details.

#### Which issue(s) this PR fixes:

Part of #94009.

#### Special notes for your reviewer:

I chose to opt-out benchmark testing.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Can you remove Service events from the "all events" list?

**Comment by Huang-Wei** [comment]
> Can you remove Service events from the "all events" list?

Added a commit to cover this. (also added an integration test to ensure ServiceAffinity still functions well)

File: test/integration/scheduler/queue_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+	// Wait for pod2 to be present in the scheduling queue.
+	if err := wait.Poll(time.Millisecond*200, wait.ForeverTestTimeout, func() (bool, error) {
+		return len(testCtx.Scheduler.SchedulingQueue.PendingPods()) == 1, nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Pop Pod2 out. It should be unschedulable.
+	podInfo := testCtx.Scheduler.NextPod()
```
Comment (alculquicondor) [inline test/integration/scheduler/queue_test.go:121]: should we expose scheduleOne instead?

Comment (Huang-Wei) [reply test/integration/scheduler/queue_test.go:121]: In very rare cases (like this one) we need that, so probably no.

Comment (alculquicondor) [reply test/integration/scheduler/queue_test.go:121]: the same could be said about `NextPod` though.

Comment (Huang-Wei) [reply test/integration/scheduler/queue_test.go:121]: 🤷

**Review by alculquicondor — APPROVED** [review]

**Pull request merged**